presto-native-execution/presto_cpp/main/connectors/PrestoToVeloxConnector.cpp

```diff
@@ -1493,6 +1493,114 @@ IcebergPrestoToVeloxConnector::createConnectorProtocol() const {
+
+std::unique_ptr<velox::connector::ConnectorInsertTableHandle>
+IcebergPrestoToVeloxConnector::toVeloxInsertTableHandle(
+    const protocol::CreateHandle* createHandle,
+    const TypeParser& typeParser) const {
+  auto icebergOutputTableHandle =
+      std::dynamic_pointer_cast<protocol::iceberg::IcebergOutputTableHandle>(
+          createHandle->handle.connectorHandle);
+
+  VELOX_CHECK_NOT_NULL(
+      icebergOutputTableHandle,
+      "Unexpected output table handle type {}",
+      createHandle->handle.connectorHandle->_type);
+
+  bool isPartitioned{false};
+  const auto inputColumns = toHiveColumns(
+      icebergOutputTableHandle->inputColumns, typeParser, isPartitioned);
+
+  return std::make_unique<
+      velox::connector::hive::iceberg::IcebergInsertTableHandle>(
+      inputColumns,
+      std::make_shared<connector::hive::LocationHandle>(
+      fmt::format("{}/data", icebergOutputTableHandle->outputPath),
+      fmt::format("{}/data", icebergOutputTableHandle->outputPath),
+          connector::hive::LocationHandle::TableType::kNew),
+      toVeloxIcebergPartitionSpec(
+          icebergOutputTableHandle->partitionSpec, typeParser),
+      toVeloxFileFormat(icebergOutputTableHandle->fileFormat),
+      nullptr,
+      std::optional(
+          toFileCompressionKind(icebergOutputTableHandle->compressionCodec)));
+}
+
+std::unique_ptr<velox::connector::ConnectorInsertTableHandle>
+IcebergPrestoToVeloxConnector::toVeloxInsertTableHandle(
+    const protocol::InsertHandle* insertHandle,
+    const TypeParser& typeParser) const {
+  auto icebergInsertTableHandle =
+      std::dynamic_pointer_cast<protocol::iceberg::IcebergInsertTableHandle>(
+          insertHandle->handle.connectorHandle);
+
+  VELOX_CHECK_NOT_NULL(
+      icebergInsertTableHandle,
+      "Unexpected insert table handle type {}",
+      insertHandle->handle.connectorHandle->_type);
+
+  bool isPartitioned{false};
+  const auto inputColumns = toHiveColumns(
+      icebergInsertTableHandle->inputColumns, typeParser, isPartitioned);
+
+  return std::make_unique<connector::hive::iceberg::IcebergInsertTableHandle>(
+      inputColumns,
+      std::make_shared<connector::hive::LocationHandle>(
+          fmt::format("{}/data", icebergInsertTableHandle->outputPath),
+          fmt::format("{}/data", icebergInsertTableHandle->outputPath),
+          connector::hive::LocationHandle::TableType::kExisting),
+      toVeloxIcebergPartitionSpec(
+          icebergInsertTableHandle->partitionSpec, typeParser),
+      toVeloxFileFormat(icebergInsertTableHandle->fileFormat),
+      nullptr,
+      std::optional(
+          toFileCompressionKind(icebergInsertTableHandle->compressionCodec)));
+}
+
+std::vector<std::shared_ptr<const connector::hive::HiveColumnHandle>>
+IcebergPrestoToVeloxConnector::toHiveColumns(
+    const protocol::List<protocol::iceberg::IcebergColumnHandle>& inputColumns,
+    const TypeParser& typeParser,
+    bool& hasPartitionColumn) const {
```

Comment:
We should avoid updating reference input parameters... Is it possible to return a pair of std::vector<std::shared_ptr<const connector::hive::HiveColumnHandle>> and hasParittionColumn ?